src/main/java/uk/gov/hmcts/ccd/data/casedataaccesscontrol/DefaultRoleAssignmentRepository.java

```diff
@@ -208,6 +218,38 @@ public RoleAssignmentResponse findRoleAssignmentsByCasesAndUsers(List<String> ca
         }
     }
 
+    private RoleAssignmentResponse findRoleAssignmentsByCasesAndUsers(RoleAssignmentQuery roleAssignmentQuery) {
+        RoleAssignmentResponse roleAssignmentResponse = new RoleAssignmentResponse();
+        roleAssignmentResponse.setRoleAssignments(new ArrayList<>());
+        final HttpHeaders headers = securityUtils.authorizationHeaders();
+
+        int pageNumber = Integer.parseInt(applicationParams.getAmRoleAssignmentStartingPageNumber());
+        int pageSize = Integer.parseInt(applicationParams.getAmRoleAssignmentPageSize());
+        int totalRecords = 0;
+
+        do {
+            headers.add(AM_PAGE_NUMBER_HEADER, Integer.toString(pageNumber));
+            headers.add(AM_PAGE_SIZE_HEADER, Integer.toString(pageSize));
+            final var requestEntity = new HttpEntity(roleAssignmentQuery, headers);
+            ResponseEntity<RoleAssignmentResponse> responseEntity = restTemplate.exchange(
+                applicationParams.amQueryRoleAssignmentsURL(),
+                HttpMethod.POST,
+                requestEntity,
+                RoleAssignmentResponse.class);
+
+            if (responseEntity.getStatusCode().is2xxSuccessful()) {
```

Comment:
not sure what happens in case of an issue at RAS returning a non-2XX. Let's discuss tomorrow.

Reply:
The code can handle it cleanly. I added few non-2xx test cases. Please review.